### PR TITLE
Add systemd notify support to auth and recursor

### PR DESCRIPTION
**Pull request by pieterlexis**

### CODE REVIEW

**Comment by pieterlexis** [comment]
@rubenk, @rgacogne comments very welcome.

File: pdns/pdns_recursor.cc [hunk 1]
```diff
@@ -2651,6 +2655,9 @@ int serviceMain(int argc, char*argv[])
   if(g_numThreads == 1) {
     L<<Logger::Warning<<"Operating unthreaded"<<endl;
     recursorThread(0);
+#ifdef HAVE_SYSTEMD
+    sd_notify(0, "READY=1");
```
Comment (rgacogne) [inline pdns/pdns_recursor.cc]: Shouldn't the call be before calling `recursorThread()`?

**Comment by rgacogne** [comment]
except for the previous comment, LGTM

**Comment by pieterlexis** [comment]
The problem is we still can exit with a config error (webserver) in recursorThread, although.. But when you're running unthreaded you're (hopefully) not doing production.

**Comment by rgacogne** [comment]
Not sure if relevant, but it might be a good idea to prevent the services files from being installed on trusty, as done in #3837 ?

**Comment by pieterlexis** [comment]
I was planning to fix that after #3837 was merged

**Pull request merged**